seq_tools/validation/c670_rg_is_paired_in_bam.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+
+"""
+    Copyright (c) 2020, Ontario Institute for Cancer Research (OICR).
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <https://www.gnu.org/licenses/>.
+    Authors:
+        Junjun Zhang <junjun.zhang@oicr.on.ca>
+        Linda Xiang <linda.xiang@oicr.on.ca>
+        Edmund Su <linda.xiang@oicr.on.ca>
+"""
+
+
+import os
+from base_checker import BaseChecker
+import subprocess
+import re
+
+
+class Checker(BaseChecker):
+    def __init__(self, ctx, metadata, skip=False):
+        super().__init__(ctx, metadata, __name__, skip=skip)
+
+    @BaseChecker._catch_exception
+    def check(self):
+        # status already set at initiation
+        if self.status:
+            return
+
+        if not self.metadata.get('read_groups'):
+            message = "Missing 'read_groups' section in the metadata JSON"
+            self.logger.info(f'[{self.checker}] {message}')
+            self.message = message
+            self.status = 'INVALID'
+            return
+    
+            
+        offending_rgs = []
+        for rg in self.metadata.get('read_groups'):
+        
+            if not rg['is_paired_end']:
+                message = "'is_pair_end' not found in readgroup : %s" % rg['submitter_read_group_id']
+                self.logger.info(f'[{self.checker}] {message}')
+                self.message = message
+                self.status = 'INVALID'
+                return
+            
+            if not rg['file_r1'].endswith('.bam'):
+                continue
+            
+            bam_file=os.path.join(self.data_dir,rg['file_r1'])
+            cmd=['samtools', 'view', '-f','128', bam_file,"|","egrep",rg['read_group_id_in_bam'].replace("'","\\'"),"-m","10","|", "wc","-l"]
```

Comment:
> Should it be `samtools view -f 1 bam_file`? https://github.com/icgc-argo/seq-tools/pull/83#discussion_r862975352

End result should be the `-f 128` as `second in pair` cannot be set without `pair` status. I suppose the only benefit `-f 128` provides over `-f 1` is if a file is submitted with mapped reads subsetted to read1 (therefore flags will not reflect missing read2). Though it will pass the converse scenario of mapped reads subsetted to read2.

Reply:
`is_paired_end=True and BAM are single-end reads?`
`is_paired_end=False and BAM are paired-end reads?`
Added the following tests to address this:
- meta_pe_bam_se:
--[json](https://github.com/icgc-argo/seq-tools/blob/c670_rg_is_paired_in_bam/tests/submissions/anon_chr1_rnaseq/anon_chr1_meta_pe_bam_se.json)
--[result](https://github.com/icgc-argo/seq-tools/blob/c670_rg_is_paired_in_bam/tests/expected_reports/anon_chr1_rnaseq/anon_chr1_meta_pe_bam_se.validation_report.INVALID.jsonl)
- meta_se_bam_pe:
-- [json](https://github.com/icgc-argo/seq-tools/blob/c670_rg_is_paired_in_bam/tests/submissions/anon_chr1_rnaseq/anon_chr1_meta_se_bam_pe.json)
-- [result](https://github.com/icgc-argo/seq-tools/blob/c670_rg_is_paired_in_bam/tests/expected_reports/anon_chr1_rnaseq/anon_chr1_meta_se_bam_pe.validation_report.INVALID.jsonl)